projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-row.component.html

```diff
@@ -121,6 +121,64 @@
     </ng-template>
     <ng-container *ngTemplateOutlet="col.visibleIndex === 0 ? treeCellTemplate : cellTemplate"></ng-container>
 </ng-template>
+<ng-container *ngIf="pinnedColumns.length > 0 && !grid.isPinningToStart">
```

Comment:
Can't these be extrated in a template that can be used for both right and left pinning?

Reply:
Sure, it's done now.